packages/app/src/resource/ToolsPage.tsx

```diff
@@ -1,76 +1,107 @@
-          showNotification({
-            color: 'red',
-            title: 'Error',
-            message: '"$push" operation timed out. Agent may be unreachable',
-            autoClose: false,
-          });
-        }
-      });
-  }, [medplum, reference]);
+      .catch((err) => showError(normalizeErrorString(err)))
+      .finally(() => setLoadingStatus(false));
+  }, [medplum, id]);
+
+  const handlePing = useCallback(
+    (formData: Record<string, string>) => {
+      const ip = formData.ip;
+      if (!ip) {
+        return;
+      }
+      setPinging(true);
+      medplum
+        .pushToAgent(reference, ip, 'PING', ContentType.PING, true)
+        .then((pingResult: string) => setLastPing(pingResult))
+        .catch((err: unknown) => showError(normalizeErrorString(err)))
+        .finally(() => setPinging(false));
+    },
+    [medplum, reference]
+  );
+
+  function showError(message: string): void {
+    showNotification({
+      color: 'red',
+      title: 'Error',
+      message,
+      autoClose: false,
+    });
+  }
 
   return (
     <Document>
-      <Title>Ping from Agent</Title>
+      <Title order={1}>Agent Tools</Title>
       <div style={{ marginBottom: 10 }}>
         Agent: <ResourceName value={reference} link />
       </div>
-      <Form onSubmit={onSubmit}>
-        <Input.Wrapper label="IP">
-          <Input name="ip" onChange={(e) => setIp(e.target.value)} value={ip} />
-        </Input.Wrapper>
-      </Form>
-      <Button
-        mt={10}
-        mb={20}
-        type="button"
-        onClick={onSubmit}
-        loading={pinging}
-        leftSection={<IconRouter size="1rem" />}
-      >
-        Ping
+      <Divider my="lg" />
+      <Title order={2}>Agent Status</Title>
+      <p>
+        Retrieve the status of the agent. This tests whether the agent is connected to the Medplum server, and the last
+        time it was able to communicate.
+      </p>
+      <Button onClick={handleStatus} loading={loadingStatus}>
+        Get Status
       </Button>
+      {!loadingStatus && status && (
+        <Table>
+          <Table.Tbody>
+            <Table.Tr>
+              <Table.Td>Status</Table.Td>
+              <Table.Td>
+                <StatusBadge status={status} />
+              </Table.Td>
+            </Table.Tr>
+            <Table.Tr>
+              <Table.Td>Last Updated</Table.Td>
+              <Table.Td>{formatDateTime(lastUpdated)}</Table.Td>
+            </Table.Tr>
+          </Table.Tbody>
+        </Table>
+      )}
+      <Divider my="lg" />
+      <Title order={2}>Ping from Agent</Title>
+      <p>
+        Send a ping command from the agent to an IP address. Use this tool to troubleshoot local network connectivity.
+      </p>
+      <Form onSubmit={handlePing}>
+        <TextInput
+          id="ip"
+          name="ip"
+          placeholder="IP Address"
+          rightSection={
+            <ActionIcon size={24} radius="xl" variant="filled" type="submit" aria-label="Ping">
```

Comment:
I think we have an unintentional UX regression after refactoring this button. Before it would show a spinner and disable the button while the ping was in progress. Now there is no indicator that a ping is in progress and you can queue up several concurrent ping requests without knowing it if you keep clicking the button.

Reply:
Ah, good catch.  Will fix